lib/src/models/multireddit.dart

```diff
@@ -3,13 +3,406 @@
+    case IconName.shopping:
+      return "shopping";
+    case IconName.sports:
+      return "sports";
+    case IconName.style:
+      return "style";
+    case IconName.tech:
+      return "tech";
+    case IconName.travel:
+      return "travel";
+    case IconName.unusualStories:
+      return "unusual stories";
+    case IconName.video:
+      return "video";
+    case IconName.emptyString:
+      return "";
+    case IconName.none:
+      return "None";
+    default:
+      throw new DRAWInternalError('IconName: $iconName is not supported');
+  }
+}
 
 /// A class which represents a Multireddit, which is a collection of
-/// [Subreddit]s. This is not yet implemented.
+/// [Subreddit]s.
+// TODO(@ckartik): Add CommentHelper.
 class Multireddit extends RedditBase with RedditBaseInitializedMixin {
+  static const String _kDisplayName = 'display_name';
+  static const String _kFrom = "from";
+  static const String _kMultiApi = 'multireddit_api';
+  static const String _kMultiredditRename = 'multireddit_rename';
+  static const String _kMultiredditUpdate = 'multireddit_update';
+  static const String _kSubreddits = "subreddits";
+  static const String _kTo = "to";
+  static const String _kVisibility = "visibility";
+  static const String _kWeightingScheme = "weighting_scheme";
+  static const int _redditorNameInPathIndex = 2;
+  static final _subredditRegExp = new RegExp(r'{subreddit}');
+  static final RegExp _userRegExp = new RegExp(r'{user}');
+  static final RegExp _multiredditRegExp = new RegExp(r'{multi}');
+
+  // TODO(@ckartik): Try to make the _data['key_color'] value null.
+  Color get keyColor => new HexColor(_data['key_color']);
+
+  /// When was this [Multireddit] created.
+  DateTime get createdUtc => new DateTime.fromMillisecondsSinceEpoch(
+      data['created_utc'].round() * 1000,
+      isUtc: true);
+
+  /// The [IconName] associated with this multireddit.
+  ///
+  /// Can be one of { artAndDesign, ask, books, business, cars, comic,
+  /// cuteAnimals, diy, entertainment, foodAndDrink, funny, games, grooming,
+  /// health, lifeAdvice, military, modelsPinup, music, news, philosophy,
+  /// picturesAndGifs, science, shopping, sports, style, tech, travel,
+  /// unusualStories, video, emptyString, none }.
+  /// If this information is not provided, will return null.
+  IconName get iconName => IconName.values.firstWhere(
+      (e) =>
+          e.toString() ==
+          ('IconName.' + _convertToCamelCase(_data['icon_name'])),
+      orElse: () => null);
+
+  List<SubredditRef> get subreddits {
+    final subredditList = [];
+    subredditList.addAll(_data['subreddits']
+        .map((subreddit) => new SubredditRef.name(reddit, subreddit['name'])));
+    return subredditList;
+  }
+
+  /// The [Map] of data associated with this multireddit.
+  Map get data => _data;
+  Map _data;
+
+  /// The [Redditor] associated with this multireddit.
+  RedditorRef get author => new RedditorRef.name(reddit, _author);
+  String _author;
```

Comment:
`_author` should be of type `RedditorRef` so we don't have to create new ones each time `author` is accessed.